rasa/shared/core/domain.py

```diff
@@ -1789,38 +1799,117 @@ def validate(mapping: Dict[Text, Any], form_name: Text, slot_name: Text) -> None
         if not isinstance(mapping, dict):
             raise InvalidDomain(
                 f"Please make sure that the slot mappings for slot '{slot_name}' in "
-                f"your form '{form_name}' are valid dictionaries. Please see "
-                f"{DOCS_URL_FORMS} for more information."
+                f"your domain are valid dictionaries. Please see "
+                f"{DOCS_URL_SLOTS} for more information."
             )
 
         validations = {
             str(SlotMapping.FROM_ENTITY): ["entity"],
             str(SlotMapping.FROM_INTENT): ["value"],
             str(SlotMapping.FROM_TRIGGER_INTENT): ["value"],
             str(SlotMapping.FROM_TEXT): [],
+            str(SlotMapping.CUSTOM): [],
         }
 
         mapping_type = mapping.get("type")
         required_keys = validations.get(mapping_type)
 
         if required_keys is None:
             raise InvalidDomain(
-                f"Your form '{form_name}' uses an invalid slot mapping of type "
+                f"Your domain uses an invalid slot mapping of type "
                 f"'{mapping_type}' for slot '{slot_name}'. Please see "
-                f"{DOCS_URL_FORMS} for more information."
+                f"{DOCS_URL_SLOTS} for more information."
             )
 
         for required_key in required_keys:
             if mapping.get(required_key) is None:
                 raise InvalidDomain(
                     f"You need to specify a value for the key "
                     f"'{required_key}' in the slot mapping of type '{mapping_type}' "
-                    f"for slot '{slot_name}' in the form '{form_name}'. Please see "
-                    f"{DOCS_URL_FORMS} for more information."
+                    f"for slot '{slot_name}'. Please see "
+                    f"{DOCS_URL_SLOTS} for more information."
                 )
 
+    @staticmethod
+    def _get_ignored_intents(
+        mapping: Dict[Text, Any], domain: Domain, active_loop_name: Text,
+    ) -> List[Text]:
+        mapping_conditions = mapping.get("conditions")
+        active_loop_match = False
+        ignored_intents = []
+
+        if mapping_conditions:
+            for condition in mapping_conditions:
+                if condition.get("active_loop") == active_loop_name:
+                    active_loop_match = True
+                    break
+
+        if active_loop_match:
+            form_ignored_intents = domain.forms[active_loop_name].get(
```

Comment:
Do we still need form level ignored intents?

Reply:
I actually only realised last week that this wasn't discussed during the implementation proposal phase - if to maintain this key in forms, move it under each slot name or under mapping conditions? I proceeded with keeping in forms, but happy to start this conversation today with main stakeholders to find out the best course of action.